src/core/mainloop/mainloop.c

```diff
@@ -1401,6 +1402,7 @@ STATIC periodic_event_item_t periodic_events[] = {
   CALLBACK(fetch_networkstatus, NET_PARTICIPANT, 0),
   CALLBACK(launch_descriptor_fetches, NET_PARTICIPANT, FL(NEED_NET)),
   CALLBACK(rotate_x509_certificate, NET_PARTICIPANT, 0),
+  CALLBACK(check_network_participation, NET_PARTICIPANT, 0),
```

Comment:
General question.

I wonder if we can avoid having this callback every 60 seconds even *when* dormant? It seems tor will wake up if our role changes (for example, becoming an HS) or if an inbound connection arrives.

So why do we have to still check every 60 seconds that we are dormant or not while tor *is dormant*?

Reply:
We don't !  These NET_PARTICIPANT callbacks are the ones that don't happen when Tor is dormant.

Reply:
Oh, or did you mean need_net?

Reply:
Ohhhhhhhhh! Yes got confused there so that callback is _not_ active when we are not participating... yes true, my bad.